applicationset/utils/webhook.go

```diff
@@ -266,25 +308,50 @@ func shouldRefreshPRGenerator(gen *v1alpha1.PullRequestGenerator, info *prGenera
 		return false
 	}
 
-	if gen.Github == nil || info.Github == nil {
-		return false
-	}
-	if gen.Github.Owner != info.Github.Owner {
-		return false
-	}
-	if gen.Github.Repo != info.Github.Repo {
-		return false
-	}
-	api := gen.Github.API
-	if api == "" {
-		api = "https://api.github.com/"
+	if gen.GitLab != nil && info.Gitlab != nil {
+		if gen.GitLab.Project != info.Gitlab.Project {
+			return false
+		}
+
+		api := gen.GitLab.API
+		if api == "" {
+			api = "https://gitlab.com/"
+		}
+
+		urlObj, err := url.Parse(api)
+		if err != nil {
+			log.Errorf("Failed to parse repoURL '%s'", api)
+			return false
+		}
+
+		if urlObj.Hostname() != info.Gitlab.APIHostname {
+			log.Debugf("%s does not match %s", api, info.Gitlab.APIHostname)
+			return false
+		}
+
+		return true
 	}
-	if !info.Github.APIRegexp.MatchString(api) {
-		log.Debugf("%s does not match %s", gen.Github.API, info.Github.APIRegexp.String())
-		return false
+
+	if gen.Github != nil && info.Github != nil {
+		if gen.Github.Owner != info.Github.Owner {
+			return false
+		}
+		if gen.Github.Repo != info.Github.Repo {
+			return false
+		}
+		api := gen.Github.API
+		if api == "" {
+			api = "https://api.github.com/"
+		}
+		if !info.Github.APIRegexp.MatchString(api) {
+			log.Debugf("%s does not match %s", api, info.Github.APIRegexp.String())
```

Comment:
Also i found a little bug. This line should log **api** variable, not **gen.Github.API**

Reply:
already fixed it

Reply:
Nice!